File: 2015/C/day_1.c
#include <stdio.h>

void part_1() {
  FILE *file = fopen("inputs/day_1_input.txt", "r");

  char line[1000];
  char indvChar = '(';
  int counter = 0;

  // Read the line into the buffer, 1000 length at a time
  while (fgets(line, sizeof(line), file) != NULL) {
    // For each character, where line is != '\0\' which is the null terminator
    // This will keep the buffer moving
    for (int i = 0; line[i] != '\0'; i++) {
      indvChar = line[i];

      if (indvChar == '(') {
        counter += 1;
      } else if (indvChar == ')') {
        counter -= 1;
      }
    }
  }
  fclose(file);

  printf("The result Floor is %d.\n", counter);
}
void part_2() {
  FILE *file = fopen("inputs/day_1_input.txt", "r");
  char indvChar = '(';
  int counter = 0;
  int posCounter = 0;

  // Read individual char at a time
  while ((indvChar = fgetc(file)) != EOF) {
    if (indvChar == '(') {
      counter += 1;
      posCounter += 1;
    } else if (indvChar == ')') {
      counter -= 1;
      posCounter += 1;
    }

    if (counter < 0) {
      break;
    }
  }
  fclose(file);

  printf("The position where Santa first entered the basement was %d.\n",
         posCounter);
}

int main() {
  part_1();
  part_2();
}

File: 2015/Python/Day_1.py
def parse_input():
    fullArray = []

    with open("Inputs/day_1_input.txt") as input_file:
        for line in input_file:
            fullArray = list(line)
    return fullArray


def part_1(list_1):
    floor = 0

    for i in list_1:
        if i == "(":
            floor += 1
        elif i == ")":
            floor -= 1

    return floor


def part_2(list_1):
    floor = 0
    counter = 0

    for i in list_1:
        counter += 1
        if i == "(":
            floor += 1
        elif i == ")":
            floor -= 1

        if floor < 0:
            break

    return counter


if __name__ == "__main__":
    list_1 = []

    list_1 = parse_input()

    print("The answer to part 1: " + str(part_1(list_1)))
    print("The answer to part 2: " + str(part_2(list_1)))
